### ISSUE TITLE
[BUG] cudf.Series not reading data properly from cupy array

### ISSUE THREAD

**Issue by yasmina-altair** [rapidsai/cudf#5611]
**Describe the bug**
When creating a cudf Series from a slice of a cupy array, the data is not selected properly. For example, for a 2D array, when selecting the first column of a cupy array, cudf Series doesn't select the first column as pandas would. 

**Steps/Code to reproduce bug**
```python
import cupy as cp
import cudf

cp_array = cp.array([[1, 2], [3, 4], [5, 6]])
print(cp_array[:, 0])
print(cudf.Series(cp_array[:, 0]))
----------------------------------------------------------------------------------------------------------------------------------------------
[1 3 5]
0    1
1    2
2    3
dtype: int64
```

**Expected behavior**
```python
import pandas as pd
import numpy as np

np_array = np.array([[1, 2], [3, 4], [5, 6]])
print(np_array[:, 0])
print(pd.Series(np_array[:, 0]))
----------------------------------------------------------------------------------------------------------------------------------------------
[1 3 5]
0    1
1    3
2    5
dtype: int64
```
**Environment overview**
 - Environment location: Docker
 - Method of cuDF install: Docker

**Comment by kkraus14** [comment]
@brandon-b-miller can you check this? This may be fixed in 0.15 already, but looks like we're not handling the stride of the cupy array.

**Comment by brandon-b-miller** [comment]
This was fixed in https://github.com/rapidsai/cudf/pull/5305. The column constructor now calls `cupy.ascontiguousarray` during `as_column`, [here.](https://github.com/rapidsai/cudf/blob/branch-0.15/python/cudf/cudf/core/column/column.py#L1279)

**Comment by kkraus14** [comment]
Perfect, fixed by #5305!
